src/computeScreenCoords.py

```diff
@@ -1,6 +1,70 @@
+import pandas as pd 
+from scipy.interpolate import LinearNDInterpolator
+from typing import List, Tuple
+
+class InterpolationType(Enum):
+    LINEAR = 1
+    RBF_LINEAR = 2
+
+def unpackEyeCoords(eyeCoords: List[List[Tuple]]) -> List[Tuple]:
+    # unpacks [[(1,2),(3,4)], [(5,6), (7,8)]] to [(1,2,3,4), (5,6,7,8)]
+    return [sum(tupList,()) for tupList in eyeCoords]
+
+def unpackScreenCoords(screenCoords):
+    screenXCoords, screenYCoords = zip(*screenCoords)
+    return (list(screenXCoords), list(screenYCoords))
+
+class LinearRegression():
+    def __init__(self, eyeCoords: List[Tuple], screenXCoords: List, screenYCoords: List):
+        df_X = pd.DataFrame(eyeCoords)
+        # TODO: restructure to perform unpacking in models instead of beforehand to account for differences in implementations
+        df_Y = pd.DataFrame(zip(screenXCoords, screenYCoords))
+        self.model = LinearRegression()
+        self.model.fit(df_X, df_Y)
+    def computeScreenCoords(self, eyeCoords):
+        df_X = pd.DataFrame(eyeCoords)
+        prediction = self.model.predict(df_X)
+        return list(prediction.itertuples(index=False, name=None))
+        
+class LinearInterpolator():
+    def __init__(self, eyeCoords: List[Tuple], screenXCoords: List, screenYCoords: List):
+        self.xInterpolator = LinearNDInterpolator(eyeCoords, screenXCoords)
+        self.yInterpolator = LinearNDInterpolator(eyeCoords, screenYCoords)
+    def computeScreenCoords(self, eyeCoords):
+        return (self.xInterpolator(eyeCoords), self.yInterpolator(eyeCoords))
+
+class RBFInterpolator():
+    # def __init__(self, eyeCoords: List[Tuple], screenXCoords: List, screenYCoords: List, kernel=None, smoothing=None):
+    #     self.kernel = kernel
+    #     self.smoothing = smoothing
+    #     self.rbf = RBFInterpolator()
     pass
 
+class Interpolator():
+    def __init(self):
+        self.interpolator = None
+
+    def calibrateInterpolator(self, eyeCoords: List[List[Tuple]], screenCoords: List[Tuple], interpType = InterpolationType.LINEAR):
+        # eyeCoords of shape [[(x1.1,y1.1),(x1.2,y1.2)], [(x2.1,y2.1, x2.2, y2.2)], etc.]
+        # screenCoords of shape [(x1,y1), (x2,y2), (x3,y3), etc]
+        if interpType == InterpolationType.LINEAR:
+            unpackedEyeCoords = unpackEyeCoords(eyeCoords)
+            unpackedXScreenCoords, unpackedYScreenCoords = unpackScreenCoords(screenCoords)
+            self.interpolator = LinearInterpolator(unpackedEyeCoords, unpackedXScreenCoords, unpackedYScreenCoords)
+        return
+    
+    def computeScreenCoords(self, eyeCoords) -> Tuple:
+        # eyeCoords of shape [x1,y1,x2,y2]
+        unpackedEyeCoords = unpackEyeCoords(eyeCoords)
+        if self.interpolator is None:
+            raise ValueError('Interpolator not calibrated yet.') # TODO: create exception class to be more accurate? not sure
+        return self.interpolator.computeScreenCoords(unpackedEyeCoords)
+
+def computeScreenCoords(eyeCoords):
+    pass
```

Comment:
Since computeScreenCoords() is the entry point of the interpolation part of the code (it's called from main.py) it would be a good idea make it call one of the interpolator models from here.

Reply:
Yeah, I wanted to discuss this point with you guys since ideally the Interpolator class can be instantiated and its own computeScreenCoords() function can be called on the object. However, I'm not sure where the instantiation will take place (in main or in this file) as of now which is why the function is still there.
I would like to delete this function and maintain the class implementation instead, but if that's not possible I will likely just have this function return the call of the interpolator object's compute function.

Reply:
I like the idea of keeping this a class implementation and potentially instantiating the class in our main, but don't know exactly how everything will work together just yet. I imagine for now we can leave this empty and once I've started integrating my part with your part @NalinGHub, then we might have a better idea of the best way to do it.

Reply:
Yeah, moving the main entry point of the computeScreenCoords functionality to an instance of an Interpolator class sounds like a good idea, then we can manage the state a lot more cleanly (i.e. which interpolation algorithm, whether or not it's been calibrated yet, etc) than with just pure functions..

Reply:
I like the idea of keeping this a class implementation and potentially instantiating the class in our main, but don't know exactly how everything will work together just yet. I imagine for now we can leave this empty and once I've started integrating my part with your part @NalinGHub, then we might have a better idea of the best way to do it.